channels/applicationinsights-channel-js/src/Statsbeat.ts

```diff
@@ -0,0 +1,203 @@
+                _networkCounter = new NetworkStatsbeat(endpoint);
+                _sender = new Sender(undefined);
+                let senderConfig = {...config};
+                senderConfig.instrumentationKey = Statsbeat.INSTRUMENTATION_KEY;
+                _sender.initialize(senderConfig, core, extensions, pluginChain, endpoint);
+                _statsbeatMetrics = {};
+                _config = config;
+                _isEnabled = _config.disableStatsbeat !== true;
+                if (_isEnabled) {
+                    _getCustomProperties();
+                    if (!_handle) {
+                        _handle = setInterval(() => {
+                            this.trackShortIntervalStatsbeats();
+                        }, Statsbeat.STATS_COLLECTION_SHORT_INTERVAL);
+                    }
+                }
+            }
+
+            _self.isInitialized = (): boolean => {
+                return !!_isEnabled;
+            }
+
+            _self.countRequest = (endpoint: string, duration: number, success: boolean) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.totalRequestCount++;
+                counter.intervalRequestExecutionTime += duration;
+                if (success === false) {
+                    counter.totalFailedRequestCount++;
+                }
+                else {
+                    counter.totalSuccesfulRequestCount++;
+                }
+            }
+
+            _self.countException = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.exceptionCount++;
+            }
+            
+            _self.countThrottle = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.throttleCount++;
+            }
+
+            _self.countRetry = (endpoint: string) => {
+                if (!_isEnabled) {
+                    return;
+                }
+                let counter: NetworkStatsbeat = _getNetworkStatsbeatCounter(endpoint);
+                counter.retryCount++;
+            }
+
+            _self.trackShortIntervalStatsbeats = (): void => {
+                _trackRequestDuration();
+                _trackRequestsCount();
+                _sendStatsbeats();
+            }
+
+            function _getCustomProperties() {
+                _cikey = _config.instrumentationKey;
+            }
+
+            function _getNetworkStatsbeatCounter(host: string): NetworkStatsbeat {
+                // Check if counter is available
+                if (_networkCounter && _networkCounter.host === host) {
```

Comment:
If the `host` doesn't match the calling endpoint, we create a new instance? When would this happen?

Reply:
They won't match if during a previous batch send Breeze redirected the response.